3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -20,12 +18,18 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 function rollTheDices() {
   // TODO Refactor this function
   const dices = [1, 2, 3, 4, 5];
-  return rollDice(1);
+  const arrayOfPromises = dices.map(el => rollDice(el))
+
+  Promise.all(arrayOfPromises).then(values => {
+    return Promise.all(values.map(r => r
+    )).then((results) => console.log('Resolved!', results))
+  }).catch((error) => console.log('Rejected!', error.message));
 }
 
 rollTheDices()
-  .then((results) => console.log('Resolved!', results))
-  .catch((error) => console.log('Rejected!', error.message));
 
 // ! Do not change or remove the code below
 module.export = rollTheDices;
+
+// reject does not stop executing of the function
+// so it does to the next promise
```

Comment:
That's right. A promise is passive, in that it can't control the task it is monitoring. A call to reject doesn't stop it.